tests/composite_pk/test_get.py

```diff
@@ -124,3 +124,12 @@ def test_lookup_errors(self):
 
     def test_get_user_by_comments(self):
         self.assertEqual(User.objects.get(comments=self.comment_1), self.user_1)
+
+    def test_get_previous_by_field(self):
+        stamp_1 = TimeStamped.objects.create(id=1)
+        stamp_2 = TimeStamped(id=2)
+        msg = "get_next/get_previous cannot be used on unsaved objects."
+        with self.assertRaisesMessage(ValueError, msg):
+            stamp_2.get_previous_by_created()
+        stamp_2.save()
+        self.assertEqual(stamp_2.get_previous_by_created(), stamp_1)
```

Comment:
```suggestion
        self.assertEqual(stamp_2.get_previous_by_created(), stamp_1)
        stamp_2.delete()
        with self.assertRaisesMessage(ValueError, msg):
            stamp_2.get_previous_by_created()
```
At this point, we can test also the case of removing the instance, since is defined only here.

Reply:
I think we can depend on unsaved and deleted instances having the same representation without needing to test that here. I do think the latest update is better, though, since this had the potential to be distracting.

> since is defined only here.

I guess this is the part I didn't follow earlier: we can freely add/delete test data while rely on the transaction rollback to restore the correct state for the next test, no?

Reply:
> > since is defined only here.
> 
> I guess this is the part I didn't follow earlier: we can freely add/delete test data while rely on the transaction rollback to restore the correct state for the next test, no?

Yes, on the contrary, it's not supposed to change the data you define once for TestCase in the `setuptestdata`.

Reply:
Ah ah right right right -- it leaves the in-memory representation on the model instances out of date. Thanks for the reminder.